src/main/scala/com/github/unchama/generic/effect/MonadThrowExtra.scala

```diff
@@ -0,0 +1,32 @@
+package com.github.unchama.generic.effect
+
+import cats.MonadError
+import cats.effect.Sync
+
+object MonadThrowExtra {
+
+  import cats.implicits._
+
+  def retryUntilSucceeds[F[_]: Sync, A](fa: F[A])(limit: Int): F[A] = {
+    require(limit >= 1)
+    def go(currentIterationCount: Int, occurredExceptions: List[Throwable]): F[A] = {
```

Comment:
```suggestion
    def go(currentIterationCount: Int, occurredExceptions: Option[Throwable]): F[A] = {
```

`Option` で良さそうです (`empty` / `.last` も `None` / `.get` に置き換えられる)